helm/observability-operator/templates/validating_job.yaml

```diff
@@ -0,0 +1,33 @@
+{{- if .Values.validationJob.enabled }}
```

Comment:
@QuantumEnigmaa I think the issue is that the command does not fail :D

See

```
kubectl -n=monitoring get pods -l=app.kubernetes.io/name=observability-operator --field-selector=status.phase=Error
No resources found in monitoring namespace.
```

Reply:
Yeah, that's what I'm working on currently : trying to have the job get into a `failed` state

Reply:
See:

```
kubectl -n=monitoring get pods -l=app.kubernetes.io/name=observability-operator --field-selector=status.phase=Error --ignore-not-found=true
> echo $?
130
> kubectl -n=monitoring get pods -l=app.kubernetes.io/name=observability-operator --field-selector=status.phase=Error
No resources found in monitoring namespace.
> echo $?
0
```

So you should use --ignore-not-found=true

Reply:
When I try it on my side I also get a 0 return code

Reply:
true :(

Reply:
Maybe you need to make it a sh script then, it errors if get is empty

Reply:
You could use jsonPath instead?

```
kubectl -n=monitoring get pods -l=app.kubernetes.io/name=observability -o jsonpath='{.items[].status.phase}'
error: error executing jsonpath "{.items[].status.phase}": Error executing template: array index out of bounds: index 0, length 0. Printing more information for debugging the template:
	template was:
		{.items[].status.phase}
	object given to jsonpath engine was:
		map[string]interface {}{"apiVersion":"v1", "items":[]interface {}{}, "kind":"List", "metadata":map[string]interface {}{"resourceVersion":""}}
```

Reply:
I've went with something of my own creation. I'm gonna see if it works but let me know what you think :)

Reply:
If that works